systems/plants/collision/Model.cpp

```diff
@@ -24,6 +24,15 @@ namespace DrakeCollision
       return nullptr;
     }
   }
+  
+  void Model::getTerrainPoints(ElementId id0, Eigen::Matrix4Xd &terrain_points) {
+    auto element_iter = elements.find(id0);
+    if ( element_iter != elements.end()) {
+      element_iter->second->getPoints(terrain_points);
+    } else {
+      terrain_points = Eigen::Matrix4Xd();
```

Comment:
This doesn't look right; not 100% sure, but I believe you need to use the `Matrix(Index rows, Index cols)` constructor with cols set to zero instead of the default constructor.